wd_helper.au3

```diff
@@ -2180,6 +2180,57 @@ Func _WD_JsonActionPause($iDuration)
+; Name ..........: _WD_JsonCookie
+; Syntax ........: _WD_JsonCookie($sName, $sValue[, $sPath = Default[, $sDomain = Default[, $bSecure = Default [,
+;                  $bHTTPOnly = Default[, $iExpiryTime = Default[, $sSameSite = Default]]]]]])
+; Parameters ....: $sName               - a string value. The name of the cookie.
+;                  $sValue              - a string value. The cookie value.
+;                  $sPath               - [optional] a string value. Default is '/'. This defines the cookie path.
+;                  $sDomain             - [optional] a string value. Default is '' - will be ommited and current browsing context’s of active document’s will be used. This defines the domain the cookie is visible to.
+;                  $bSecure             - [optional] a boolean value. Default is False. This defines whether the cookie is a secure cookie.
+;                  $bHTTPOnly           - [optional] a boolean value. Default is False. This defines whether the cookie is an HTTP only cookie.
+;                  $iExpiryTime         - [optional] an integer value. Default is 0 - will be ommited. This defines when the cookie expires, specified in seconds since Unix Epoch.
+;                  $sSameSite           - [optional] a string value. Default will be ommited and this mean "None". This defines whether the cookie applies to a SameSite policy. One of the following modes:
+;                  |None
+;                  |Lax
+;                  |Strict
+; Return values .: cookie as JSON strings
+; Author ........: mLipok
+; Modified ......:
+; Remarks .......:
+; Related .......: _WD_Cookies
+; Link ..........: https://www.w3.org/TR/webdriver/#dfn-table-for-cookie-conversion
+; Example .......: No
+; ===============================================================================================================================
+Func _WD_JsonCookie($sName, $sValue, $sPath = Default, $sDomain = Default, $bSecure = Default , $bHTTPOnly = Default, $iExpiryTime = Default, $sSameSite =Default)
+	Local Const $sFuncName = "_WD_JsonCookie"
+	If $sPath = Default Then $sPath = '/'
+	If $sDomain = Default Then $sDomain = ''
+	If $bSecure = Default Then $bSecure = False
+	If $bHTTPOnly = Default Then $bHTTPOnly = False
+	If $iExpiryTime = Default Then $iExpiryTime  = 0
+	If $sSameSite = Default Then $sSameSite = ''
+
+	Local $vData = Json_ObjCreate()
+	Json_Put($vData, '.cookie.name', $sName)
+	Json_Put($vData, '.cookie.value', $sValue)
+	If $sPath Then Json_Put($vData, '.cookie.path', $sPath)
+	If $sDomain Then Json_Put($vData, '.cookie.domain', $sDomain)
+	If $bSecure Then Json_Put($vData, '.cookie.secure', $bSecure)
+	If $bHTTPOnly Then Json_Put($vData, '.cookie.httponly', $bHTTPOnly)
+	If $iExpiryTime Then Json_Put($vData, '.cookie.expiry', $iExpiryTime)
+	If $sSameSite Then Json_Put($vData, '.cookie.sameSite', $sSameSite)
```

Comment:
I believe this would be better implemented if you did it like this --

-  Eliminate the first section of `If $variable = Default` checks
- Change the second section of If statements to `If $variable <> Default`

This will allow for doing things like passing False for $bSecure and having it correctly added to the cookie

Reply:
Take a look: [_WD_JsonCookie() inner validation](https://github.com/Danp2/au3WebDriver/pull/194/commits/6e4896a09377b54c2a51ed96ebef4576772545b5)

Will this be fine in this way ?

Reply:
The idea was that the optional parameters didn't generate any entries unless they were supplied. This change doesn't accomplish that.

Reply:
Default in the meaning of WebDriver spec is to omnit so you check `If $variable <> Default` and `Then Json_Put($vData, '.cookie.`
This way it works like intended by WebDriver spec .
But in this case (your modification) this is user resposibility to provide valid data.

And this is your intentionall choices of how this functions should works?

Reply:
> And this is your intentionall choices of how this functions should works?

If so, I can agree/aprove this way.

Reply:
> This way it works like intended by WebDriver spec

Unsure what you mean by this. 

IMO, this UDF isn't responsible for supplying default values to match the W3C specs. These entries are optional. When missing, it the webdriver's responsibility to supply any default values.

> But in this case (your modification) this is user resposibility to provide valid data.

As I said, additional checking for variable types could be added if desired. 

> And this is your intentionall choices of how this functions should works?

This is your function. I'm simply suggesting what I believe would be a better implementation where the `.cookie` entries will only be added for user-supplied values.

Reply:
> > This way it works like intended by WebDriver spec
> 
> Unsure what you mean by this.
> 
> IMO, this UDF isn't responsible for supplying default values to match the W3C specs. These entries are optional. When missing, it the webdriver's responsibility to supply any default values.

I mean that https://www.w3.org/TR/webdriver/#dfn-table-for-cookie-conversion says that if this parameters are omnited then WebDriver use the defaults by itself.

Reply:
Right.... those defaults get applied by the webdriver in the event the entry is omitted. Therefore, it isn't necessary for _WD_JsonCookie to handle these defaults.

Agreed?

Reply:
> This is your function. 

Oh indeed.
But you are the captain on this ship.

> I'm simply suggesting what I believe would be a better implementation where the `.cookie` entries will only be added for user-supplied values.

And here you use your captain's privileges.

Overall, it doesn't matter ....
It is important that I agree with your implementation / modifications.